### ISSUE TITLE
Provide human readable names for model labels

### ISSUE THREAD

**Issue by saurabheights** [ultralytics/ultralytics#4092]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

This is a follow up request of issue https://github.com/ultralytics/ultralytics/issues/3563

Kindly consider using more human-readable names, especially for popular or publicly available models (for e.g. `RTDETR` is taken from Ultralytics documentation, so I would consider that as a more public/popular model). This practice can help reduce the likelihood of client-side bugs and facilitate better interactions. Implementing such changes at the source level (upstream) would be more efficient.

### Use case

I am currently using Ultralytics for multi-video multi-object detection.  I want to add option for user to select a publicly available models based on the labels they need. The problem is displaying the classes as 0, 1, 2, or 3 is not the most user-friendly solution. It would be more ideal to use descriptive names for the classes.

Also, a dict of {0:0, 1:1} doesnt convey any extra information. If we already know the number of classes, I will know what this dict contain.

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@saurabheights thank you for your suggestion and for your willingness to contribute to the improvement of Ultralytics YOLOv8. 

We understand that using more descriptive, human-readable labels for classes can provide a more user-friendly experience, especially for those unfamiliar with the structure of the models. 

We appreciate your feedback and recognize the potential development considerations, particularly given your use case of multi-object detection across multiple videos. It would indeed be more useful to implement meaningful labels, rather than numerical values, to describe the classes.

Here's how we might proceed: Rather than labeling classes as 0, 1, 2, 3, we could use labels that describe the class of objects to be detected, such as 'person', 'car', 'dog', etc. This would require mapping each numerical class label to a descriptive label. 

However, please note that these labels can vary between different datasets. A class value '0' in one dataset might correspond to 'person', while in another dataset it might correspond to a different object. Therefore, it will be important to ensure that the mapping of these labels is done appropriately for each specific dataset.

We're looking forward to your proposed solution in the form of a PR. Your input is greatly appreciated and we're excited to continue improving the tool based on valuable feedback like yours.

**Comment by saurabheights** [comment]
@glenn-jocher Hi, hope I didnt respond too late. My major concern is during inference.

> This would require mapping each numerical class label to a descriptive label. However, please note that these labels can vary between different datasets. 

We already have labels in other models such as `yolov8n.pt` (shown below in code snippet). In case I am missing something here, we can say that we have inconsistency.

```
model = YOLO('yolov8n.pt')
print(model.names)

{0: 'person', 1: 'bicycle', 2: 'car', 3: 'motorcycle', 4: 'airplane', 5: 'bus', 6: 'train', 7: 'truck', 8: 'boat', 9: 'traffic light', ..., , 79: 'toothbrush'}

# vs
model = RTDETR("rtdetr-l.pt")
print(model.names)

{0: '0', 1: '1', 2: '2', 3: '3', ... }
```

I dived deeper into ultralytics code and it looks like `yolov8n.pt` model is saved via [pytorch save-load-entire-model api](https://pytorch.org/tutorials/beginner/saving_loading_models.html#save-load-entire-model). This probably had `model.names` set to descriptive names before saving and thus are available upon loading for all your users, but that was not the case for RTDETR models. 

> However, please note that these labels can vary between different datasets. A class value '0' in one dataset might correspond to 'person', while in another dataset it might correspond to a different object. Therefore, it will be important to ensure that the mapping of these labels is done appropriately for each specific dataset.

I completely agree. Its very easy to refer to wrong source for labels and hence care must be taken by each user when using pretrained model. Ideally, this work can be easily reduced by library maintainers when a model weight file is being uploaded for user consumption.

IMO, public models checkpoints file which don't have the names variable set, should be fixed. I believe just loading the model, setting name dictionary and saving them back should be enough to fix existing models (I could be wrong). Ofcourse, correct me if I am wrong here. Furthermore, for new public models, care must be taken via automated tests that the labels are set.

```
# Just a quick test code to copy labels from yolov8n.pt to rtdetr-l.pt
model1 = YOLO('yolov8n.pt')
model2 = RTDETR("rtdetr-l.pt")
model2.model.names = model1.model.names
torch.save(model2, "rtdetr-l-names.pt")
model3 = torch.load("rtdetr-l-names.pt")
print(model3.names)
```

Please do let me know your feedback and if there is any way I can assist here.
